p2p/types/peers_view.go

```diff
@@ -113,3 +114,20 @@ func (view *sortedPeersView) sortAddrs(startAddr crypto.Address) {
 	}
 	view.sortedAddrs = append(view.sortedAddrs[i:len(view.sortedAddrs)], view.sortedAddrs[0:i]...)
 }
+
+// getAddrIndex returns the sortedAddrs index at which the given address is stored
+// or at which to insert it if not present.
+func (view *sortedPeersView) getAddrIndex(addr crypto.Address) int {
+	wrapIdx := sort.Search(len(view.sortedAddrs), func(visitIdx int) bool {
+		return view.sortedAddrs[visitIdx] < view.sortedAddrs[0]
+	})
+
+	frontAddrs := view.sortedAddrs[:wrapIdx]
+	backAddrs := view.sortedAddrs[wrapIdx:]
+	i := sort.SearchStrings(frontAddrs, addr.String())
+	if i == 0 {
```

Comment:
Here I would add a simple comment explaining the "why" of this.
At a glance, I am a bit confused because I remember that `SearchString` returns `len(a)` when something is not found and that's, generally speaking, the most common check that's done after searching for a needle in a haystack.

Wdyt?

Reply:
:100: perfect suggestion - thank you! :raised_hands: 

You're correct about `SearchString()`s return value, in context, index `0` means "before the self addr". In that case we need to consider `backAddrs` because, by definition, all of its values are "smaller" than self addr.
_(mostly writing this for myself to work out how to phrase the comment - suggestions are welcome :stuck_out_tongue_closed_eyes:)_